Ticket Booking System/Business/Booking.cs

```diff
@@ -59,5 +59,26 @@ private JourneyStatus SetjourneyStatus()
                 return JourneyStatus.MultiCity;
             return JourneyStatus.RoundTrip;
         }
+        public bool Compare(Booking booking)
+        {
+            var isValid = this.tickets.Where(ticket => 
+            booking.tickets.All(ticket2=>
+            ticket.Compare(ticket2))).Count().Equals(tickets.Count());
+            if (!isValid)
+                return false;
+            if (!booking.journeyStatus.Equals( this.journeyStatus )&& booking.journeyStatus != null)
```

Comment:
```suggestion
if (booking.journeyStatus != null && !booking.journeyStatus.Equals(this.journeyStatus))
```

also please apply this for all the if statements below.